public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -358,9 +358,15 @@ function (angular, _, moment, dateMath, kbn, CloudWatchAnnotationQuery) {
     }
 
     this.getExpandedVariables = function(target, dimensionKey, variable) {
+      /* if the all checkbox is marked we should add all values to the targets */
+      var allSelected = _.find(variable.options, {'selected': true, 'text': 'All'});
```

Comment:
When selecting all, the variable.current.value contains $__all if that can easily be expanded to all the values the code can be made better.